### PR TITLE
feat: accept vectors of size 2 in addition to MapEntry in unparse

### LINKED ISSUE #1123: Clarify and maybe change `unparse` behavior
As posted on [slack](https://clojurians.slack.com/archives/CLDK6MFMK/p1731615638557769), data equal to the result of `malli.core/parse`  cannot be unparsed.

``` clojure
(def int-or-string [:orn [:int :int] [:str :string]])

(malli.core/parse int-or-string 1) ;; -> [:int 1]

(= [:int 1] (malli.core/parse int-or-string 1)) ;; -> true

(malli.core/unparse int-or-string [:int 1]) ;=> :malli.core/invalid

(->> 1
     (malli.core/parse int-or-string)
     (malli.core/unparse int-or-string)) ;; => 1
```
I have some data in my schema's parsed format, which was not produced by `parse`, but which I would like to unparse. It appears that's not currently possible.

### PR DESCRIPTION
**Pull request by opqdonut**
fixes #1123

### CODE REVIEW

File: src/malli/core.cljc [hunk 1]
```diff
@@ -872,8 +872,8 @@
             (let [unparsers (into {} (map (fn [[k _ c]] [k (-unparser c)])) (-children this))]
               (fn [x]
                 (if (miu/-tagged? x)
-                  (if-some [unparse (get unparsers (key x))]
-                    (unparse (val x))
+                  (if-some [unparse (get unparsers (first x))]
+                    (unparse (second x))
```
Comment (frenchy64) [inline src/malli/core.cljc:876]: `nth` might be more equivalent in performance here since I think `first` needs to go via `seq`.

**Comment by ikitommi** [comment]
```clojure
(def schema
  [:or
   [:tuple :string :keyword]
   [:orn
    ["int" :int]
    ["any" :keyword]]])

(->> (m/parse schema :any) (m/unparse schema))
; => ["any" :any]
```

**Comment by opqdonut** [comment]
Ouch, that's a nice corner-case! This behaviour is the same in `master` since a `MapEntry` is a `:tuple`. The safest solution would be to switch to something like a custom record, but that wouldn't help users solve the original problem (of fabricating inputs to unparse easily).

```clj
;; in master
(m/validate [:tuple :string :keyword] (clojure.lang.MapEntry. "a" :a)) ; => true
```

**Comment by ikitommi** [comment]
Indeed. So, to get correct behavior, we would  need the record. Maybe for the `:catn` case too where we use regular map now? If the record would implement right interfaces / protocols, this would not even be a breaking change?

**Comment by ikitommi** [comment]
.. but would this solve the original issue?

**Comment by ikitommi** [comment]
```clojure
(defrecord Tagged [key value])

(defn -tagged [key value] (->Tagged key vvalue))
(defn -tagged? [x] (instance? Tagged x))
```
+ changing access to `(:key tagged)` and `(:value tagged)`:

```clojure
(def schema
  [:or
   [:tuple :string :keyword]
   [:orn ["any" :keyword]]])

(->> :any (m/parse schema))
; => #malli.impl.util.Tag{:key "any", :value :any}

(m/unparse schema *1)
; => :any

(m/unparse schema (miu/-tagged "any" :any))
; => :any
```

I think this would solve this and no ambiguity with vectors.

**Comment by opqdonut** [comment]
The record wouldn't fully solve the original problem (making custom inputs to unparse easy), but would at least solve the footgun (things look like vectors but aren't). I agree that the record would be the most correct solution.

Re: breaking, I think it would be breaking for people who are consuming the output of `parse`.

**Comment by ikitommi** [comment]
> Re: breaking, I think it would be breaking for people who are consuming the output of parse.

true that. Tried to look if the parse syntax is agreed to be stable or not, no mention of that so user can assume it is, thus breaking.

**Pull request closed without merging**